Tests/SwiftSyntaxMacroExpansionTest/MacroReplacementTests.swift

```diff
@@ -113,4 +114,52 @@ final class MacroReplacementTests: XCTestCase {
       """
     )
   }
+
+  func testMacroGenericArgumentExpansion() throws {
+    let macro: DeclSyntax =
+      """
+      macro gen<A, B>(a: A, b: B) = #otherMacro<A, B>(first: a, second: b)
+      """
+
+    let use: ExprSyntax =
+      """
+      #gen<Int, String>(a: 5, b: "Hello")
+      """
+
+    let macroDecl = macro.as(MacroDeclSyntax.self)!
+    let definition = try macroDecl.checkDefinition()
+    guard case let .expansion(expansion, replacements, genericReplacements) = definition else {
+      XCTFail("not a normal expansion")
+      fatalError()
+    }
+
+    guard let replacementA = genericReplacements.first else {
+      XCTFail("Expected generic replacement for A")
+      fatalError()
+    }
```

Comment:
We should not `fatalError` in test cases because that takes down the entire test suite when it fails. A `return` would be better. Or, even better use `XCTUnwrap` where possible.

```suggestion
    let replacementA = try XCTUnwrap(genericReplacements.first)
```

Same below and above.

Reply:
Ok, sure - i was copying the style this file seems to be doing. I can return instead.